### PR TITLE
feat(preset): Add NestJs Package Group rule and update Nestjs Monorepo definition

### PR DESCRIPTION
**Pull request by Xyaren**
## Changes

Added several NestJs Repositories similar to #16520.

## Context

Nestjs dependencies are updateable on their own most times.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

**Comment by SchroederSteffen** [comment]
I was also thinking about adding more repositories. I hesitated to do so, because they use different major versions and I wasn't sure if this causes issues. E.g. `@nestjs/swagger` is on `6.x` while the core packages are on `9.x`.

Instead of listing them all, we could also just use the prefix `https://github.com/nestjs`, no?

**Comment by Xyaren** [comment]
Although they are using different major versions they are quite strong coupled. Currently I have two separate Merge Requests from Renovate which are both failing, one for nest-monorepo, one for swagger.

**Review by viceice — CHANGES_REQUESTED** [review]
I think they should only be grouped for major updates then. something like we to with ts-jest.
https://github.com/renovatebot/renovate/blob/4953abf7caccaf0b07ea708550e33e6b6526e47b/lib/config/presets/internal/group.ts#L642-L651

**Comment by Xyaren** [comment]
Should I do this for all the nestjs modules ?

**Comment by viceice** [comment]
yes, the first one

**Comment by Xyaren** [comment]
Should I remove the nestsjs monorepo defenition then ? 
The group would cover all the modules already right ?

**Comment by viceice** [comment]
If the core newjs repo published multiple packages, it should stay as a single monorepo

File: lib/config/presets/internal/group.ts [hunk 1]
```diff
@@ -659,6 +659,16 @@ const staticGroups = {
       },
     ],
   },
+  nestjsMajor: {
```
Comment (viceice) [inline lib/config/presets/internal/group.ts:662]: i think you also need to update the `groupRecommend` preset above

Comment (viceice) [reply lib/config/presets/internal/group.ts:662]: ☝️

File: lib/config/presets/internal/monorepo.ts [hunk 2]
```diff
@@ -144,12 +144,7 @@ const repoGroups = {
   'mikro-orm': 'https://github.com/mikro-orm/mikro-orm',
   mockito: 'https://github.com/mockito/mockito',
   mstest: 'https://github.com/microsoft/testfx',
-  nest: [
-    'https://github.com/nestjs/nest',
-    'https://github.com/nestjs/passport',
-    'https://github.com/nestjs/schematics',
-    'https://github.com/nestjs/terminus',
```
Comment (rarkins) [inline lib/config/presets/internal/monorepo.ts:151]: Why not group these anymore?

File: lib/config/presets/internal/group.ts [hunk 3]
```diff
@@ -659,6 +659,16 @@ const staticGroups = {
...
+    description: 'Group `@nestjs/*` packages major version updates',
+    packageRules: [
+      {
+        matchSourceUrlPrefixes: ['https://github.com/nestjs/'],
+        matchUpdateTypes: ['major'],
```
Comment (rarkins) [inline lib/config/presets/internal/group.ts:667]: Why only group if the updates are major?

**Review by rarkins — CHANGES_REQUESTED** [review]

**Pull request closed without merging**